packages/components/src/utils/getTextAsHtml.ts

```diff
@@ -20,31 +22,79 @@ export const getTextAsHtml = (content?: (HardBreakProps | TextProps)[]) => {
     return ""
   }
 
-  return content
-    .map((node) => {
-      if (node.type === "hardBreak") {
-        return "<br />"
-      }
+  const output: string[] = []
+  let existingLinkMark: Marks | undefined = undefined
+
+  // At every step, we will close off all marks except for links
+  // First encounter with a link, always open it first before other marks
+  // Close all other marks first before closing the link mark
+  content.forEach((node) => {
+    if (node.type === "hardBreak") {
+      output.push("<br />")
+      return
+    }
+
+    const newLinkMark = node.marks?.find((mark) => mark.type === "link")
+    const isLinkMarkNew =
+      !!newLinkMark && !isEqual(existingLinkMark, newLinkMark)
```

Comment:
Clarified offline, strict equality is important here because as long as the values are different, the functionality will definitely be different alr (cos there is only `target` and `href`).